Plutarch/Internal.hs

```diff
@@ -161,27 +170,59 @@ plam' f = Term $ \i ->
+    getArityBuiltin (RBuiltin PLC.SubtractInteger) = Just 1
+    getArityBuiltin (RBuiltin PLC.MultiplyInteger) = Just 1
+    getArityBuiltin (RBuiltin PLC.DivideInteger) = Just 1
+    getArityBuiltin (RBuiltin PLC.QuotientInteger) = Just 1
+    getArityBuiltin (RBuiltin PLC.RemainderInteger) = Just 1
+    getArityBuiltin (RBuiltin PLC.ModInteger) = Just 1
+    getArityBuiltin (RBuiltin PLC.EqualsInteger) = Just 1
+    getArityBuiltin (RBuiltin PLC.LessThanInteger) = Just 1
+    getArityBuiltin (RBuiltin PLC.LessThanEqualsInteger) = Just 1
+    getArityBuiltin (RBuiltin PLC.AppendByteString) = Just 1
+    getArityBuiltin (RBuiltin PLC.ConsByteString) = Just 1
+    getArityBuiltin (RBuiltin PLC.SliceByteString) = Just 2
+    getArityBuiltin (RBuiltin PLC.LengthOfByteString) = Just 0
+    getArityBuiltin (RBuiltin PLC.IndexByteString) = Just 1
+    getArityBuiltin (RBuiltin PLC.EqualsByteString) = Just 1
+    getArityBuiltin (RBuiltin PLC.LessThanByteString) = Just 1
+    getArityBuiltin (RBuiltin PLC.LessThanEqualsByteString) = Just 1
+    getArityBuiltin (RBuiltin PLC.Sha2_256) = Just 0
+    getArityBuiltin (RBuiltin PLC.Sha3_256) = Just 0
+    getArityBuiltin (RBuiltin PLC.Blake2b_256) = Just 0
+    getArityBuiltin (RBuiltin PLC.VerifySignature) = Just 2
+    getArityBuiltin (RBuiltin PLC.AppendString) = Just 1
+    getArityBuiltin (RBuiltin PLC.EqualsString) = Just 1
+    getArityBuiltin (RBuiltin PLC.EncodeUtf8) = Just 0
+    getArityBuiltin (RBuiltin PLC.DecodeUtf8) = Just 0
+    getArityBuiltin (RForce (RBuiltin PLC.IfThenElse)) = Just 2
+    getArityBuiltin (RForce (RBuiltin PLC.ChooseUnit)) = Just 1
+    getArityBuiltin (RForce (RBuiltin PLC.Trace)) = Just 1
+    getArityBuiltin (RForce (RForce (RBuiltin PLC.FstPair))) = Just 0
+    getArityBuiltin (RForce (RForce (RBuiltin PLC.SndPair))) = Just 0
+    getArityBuiltin (RForce (RForce (RBuiltin PLC.ChooseList))) = Just 2
+    getArityBuiltin (RForce (RBuiltin PLC.MkCons)) = Just 1
+    getArityBuiltin (RForce (RBuiltin PLC.HeadList)) = Just 0
+    getArityBuiltin (RForce (RBuiltin PLC.TailList)) = Just 0
+    getArityBuiltin (RForce (RBuiltin PLC.NullList)) = Just 0
+    getArityBuiltin (RForce (RBuiltin PLC.ChooseData)) = Just 5
+    getArityBuiltin (RBuiltin PLC.ConstrData) = Just 1
+    getArityBuiltin (RBuiltin PLC.MapData) = Just 0
+    getArityBuiltin (RBuiltin PLC.ListData) = Just 0
+    getArityBuiltin (RBuiltin PLC.IData) = Just 0
+    getArityBuiltin (RBuiltin PLC.BData) = Just 0
+    getArityBuiltin (RBuiltin PLC.UnConstrData) = Just 0
+    getArityBuiltin (RBuiltin PLC.UnMapData) = Just 0
+    getArityBuiltin (RBuiltin PLC.UnListData) = Just 0
+    getArityBuiltin (RBuiltin PLC.UnIData) = Just 0
+    getArityBuiltin (RBuiltin PLC.UnBData) = Just 0
+    getArityBuiltin (RBuiltin PLC.EqualsData) = Just 1
+    getArityBuiltin (RBuiltin PLC.MkPairData) = Just 1
+    getArityBuiltin (RBuiltin PLC.MkNilData) = Just 0
+    getArityBuiltin (RBuiltin PLC.MkNilPairData) = Just 0
```

Comment:
There probably is a way to have `getArityBuiltin` with new plutus (and even infer builtin function types from TypeScheme); I will look at it when I get a chance.

Reply:
The issue is that Plutus internals change. Do we really want to depend on the internals?

Reply:
We can stop relying on the internal functions, and use plutus' types/constraints only (like `TypeScheme`), which I don't imagine will change often.